Common/Orders/TimeInForces/DayTimeInForceHandler.cs

```diff
@@ -0,0 +1,92 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Orders.TimeInForces
+{
+    /// <summary>
+    /// Handles the Day time in force for an order (DAY)
+    /// </summary>
+    public class DayTimeInForceHandler : ITimeInForceHandler
+    {
+        /// <summary>
+        /// Checks if an order has expired
+        /// </summary>
+        /// <param name="security">The security matching the order</param>
+        /// <param name="order">The order to be checked</param>
+        /// <returns>Returns true if the order has expired, false otherwise</returns>
+        public bool HasOrderExpired(Security security, Order order)
+        {
+            var exchangeHours = security.Exchange.Hours;
+
+            var orderTime = order.Time.ConvertFromUtc(exchangeHours.TimeZone);
+            var time = security.LocalTime;
+
+            bool expired;
+            switch (order.SecurityType)
+            {
+                case SecurityType.Forex:
+                case SecurityType.Cfd:
+                    // With real brokerages (IB, Oanda, FXCM have been verified) FX orders expire at 5 PM NewYork time.
+                    // For now we use this fixed cut-off time, in future we might get this value from brokerage models,
+                    // to support custom brokerage implementations.
+
+                    var cutOffTimeZone = TimeZones.NewYork;
+                    var cutOffTimeSpan = TimeSpan.FromHours(17);
+
+                    orderTime = order.Time.ConvertFromUtc(cutOffTimeZone);
+                    var expiryTime = orderTime.Date.Add(cutOffTimeSpan);
+                    if (orderTime.TimeOfDay > cutOffTimeSpan)
+                    {
+                        // order submitted after 5 PM, expiry on next date
+                        expiryTime = expiryTime.AddDays(1);
+                    }
+
+                    expired = time.ConvertTo(exchangeHours.TimeZone, cutOffTimeZone) >= expiryTime;
```

Comment:
as a reminder, we discussed using `exchangeHours.GetNextMarketClose(orderTime, false)` to determine when the end of 'day' should be.

Reply:
you're 100% correct, it won't work that way -- let's keep it as you have it. If we ever need a more dynamic value we can look into sourcing the value through the brokerage model.